x-pack/test/apm_api_integration/tests/dependencies/top_spans.spec.ts

```diff
@@ -0,0 +1,235 @@
+          });
+
+          expect(omit(goSpans[0], 'traceId', 'transactionId')).to.eql({
+            '@timestamp': 1609459200000,
+            agentName: 'go',
+            duration: 50000,
+            serviceName: 'go',
+            spanName: '/_search',
+            transactionName: 'GET /api/my-other-endpoint',
+            transactionType: 'request',
+          });
+        });
+      });
+
+      describe('with a kuery', () => {
+        it('returns the correct spans for the requested spanName', async () => {
+          const response = await callApi({
+            backendName: 'elasticsearch',
+            spanName: '/_search',
+            kuery: 'service.name:go',
+          });
+
+          const { spans } = response.body;
+
+          const spanNames = uniq(spans.map((span) => span.spanName));
+          const javaSpans = spans.filter((span) => span.serviceName === 'java');
+          const goSpans = spans.filter((span) => span.serviceName === 'go');
+
+          expect(spanNames).to.eql(['/_search']);
+
+          expect(javaSpans.length).to.be(0);
+          expect(goSpans.length).to.be.greaterThan(0);
+          expect(goSpans.length).to.eql(spans.length);
+        });
+      });
+
+      describe('with an environment', () => {
+        it('returns the correct spans for the requested spanName', async () => {
+          const response = await callApi({
+            backendName: 'elasticsearch',
+            spanName: '/_search',
+            environment: 'development',
+          });
+
+          const { spans } = response.body;
+
+          const spanNames = uniq(spans.map((span) => span.spanName));
+          const javaSpans = spans.filter((span) => span.serviceName === 'java');
+          const goSpans = spans.filter((span) => span.serviceName === 'go');
+
+          expect(spanNames).to.eql(['/_search']);
+
+          expect(javaSpans.length).to.be(0);
+          expect(goSpans.length).to.be.greaterThan(0);
+          expect(goSpans.length).to.eql(spans.length);
+        });
+      });
+
+      describe('when requesting spans without a transaction', () => {
+        it('should return the spans without transaction metadata', async () => {
+          const response = await callApi({
+            backendName: 'elasticsearch',
+            spanName: 'without transaction',
+          });
+
+          const { spans } = response.body;
+
+          const spanNames = uniq(spans.map((span) => span.spanName));
+
+          expect(spanNames).to.eql(['without transaction']);
+
+          expect(omit(spans[0], 'traceId')).to.eql({
+            '@timestamp': 1609459200000,
+            agentName: 'java',
+            duration: 200000,
+            serviceName: 'java',
+            spanName: 'without transaction',
+          });
+
+          expect(spans[0].transactionType).not.to.be.ok();
+          expect(spans[0].transactionId).not.to.be.ok();
+          expect(spans[0].transactionName).not.to.be.ok();
```

Comment:
TIL: `ok()` in chai is equivalent of `truthy()` in jest. I know which one I prefer :)

Reply:
This read much better imo:

```suggestion
          expect(spans[0].transactionType).to.be.undefined();
          expect(spans[0].transactionId).to.be.undefined();
          expect(spans[0].transactionName).to.be.undefined();
```

Is that what we are asserting?

Reply:
Ahhh.. this is using should.js. Probably `to.be.undefined` doesn't exist so nvm
But wow... should.js is ANCIENT. It's not seen a commit in 8 years.

Reply:
Off topic but it might be beneficial to start using jest's expectation library for new api tests, and then slowly migrate over existing api tests.

There is a slack thread about it here: https://elastic.slack.com/archives/C010ZGLH0SF/p1654601423697289

Reply:
I wouldn't mind that, jest's matchers are pretty nice. Not sure what that means for our snapshot tests though (`expectSnapshot`).